brave/brave-ios
# Fix #6706: Add support of SNS navigation.

## Summary of Changes
Brave will now detect user input url is SNS domain
User will need to choose either disable SNS resolver or enable it. 
Leave it no chosen will be considered as getting `asked` every time. 
The settings is available under Web3. The default value is `Ask`
However, everything is behind a feature flag `WalletFeatureFlags.SNSDomainResolverEnabled`.
Local build can change this value enable to try it out.


This pull request fixes #6706

## Submitter Checklist:


## Test Plan:

1. Make sure the feature flag is on
2. Input `onybose.sol` in the omnibox then enter to go to
3. An interstitial page will show up explains what Brave can help with the SNS domain
4. Check copies are correct
5. Click `Disable`
6. observe web page will load and gives an error
7. Go to Settings/Web3/Web3 Domains
8. observe the `Resolve Solana Name Service (SNS) Domain Names` value is `Disabled`
9. chose `Ask` for `Resolve Solana Name Service (SNS) Domain Names`
10. back to the browser and hit reload 
11. observe the interstitial page is back
12. this time click `Proceed`
13. observe the url `onybose.sol` is resolved to `anirudha.co` and loads successfully 
14. Go to Settings/Web3/Web3 Domains
15. observe now he `Resolve Solana Name Service (SNS) Domain Names` value is `Enabled`

## Screenshots:


![simulator_screenshot_4E340146-77CF-4743-99CA-A204FA9DC998](https://user-images.githubusercontent.com/1187676/217322039-c172b59a-c0b2-43bf-839e-eaa02b1e0bf5.png)

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
